GeeksCoreLibrary/Components/Configurator/Configurator.cs

```diff
@@ -159,15 +137,35 @@ public override async Task<HtmlString> InvokeAsync(DynamicContent dynamicContent
                 return new HtmlString("");
             }
 
-            var mainStepCount = Settings.ComponentMode == ComponentModes.Vue ? 0 : 1;
+            Logger.LogDebug("Configurator - Mode set to: {ComponentMode}", Settings.ComponentMode);
+
+            var resultHtml = new StringBuilder();
+            switch (Settings.ComponentMode)
+            {
+                case ComponentModes.Default:
+                    resultHtml.Append(await HandleDefaultModeAsync());
+                    break;
+                case ComponentModes.Vue:
+                    resultHtml.Append(await HandleVueModeAsync());
+                    break;
+                default:
+                    throw new NotImplementedException($"Unknown or unsupported component mode '{Settings.ComponentMode}' in 'InvokeAsync'.");
```

Comment:
Can you use `ArgumentOutOfRangeException` here please? That is our new standard for exceptions of invalid switch statement cases and is also what Microsoft uses for that.

Reply:
It was based on how this was done in other components, so this needs to be fixed in other components as well (like the Account component).